dev/PushNotifications/PushManager.cpp

```diff
@@ -4,11 +4,97 @@
 #include "pch.h"
 #include "PushManager.h"
 #include "PushManager.g.cpp"
+#include <winrt\windows.networking.pushnotifications.h>
+#include <winerror.h>
+#include <algorithm>
+#include <vector>
+#include <iostream>
+#include <ChannelResult.h>
+#include <winerror.h>
+#include <PushNotificationErrorCodes.h>
+
+using namespace winrt::Windows::Networking::PushNotifications;
 
 namespace winrt::Microsoft::ProjectReunion::implementation
 {
+    std::vector<winrt::guid> PushManager::s_remoteIdList;
+    std::mutex PushManager::s_mutex;
+    std::unique_lock<std::mutex> PushManager::s_lock (s_mutex, std::defer_lock); // Avoid locking during the constructor call
+    inline constexpr std::uint32_t c_maxBackoffSeconds{ 960 };
+
+    bool PushManager::isChannelRequestRetryable(const winrt::hresult& hrException)
+    {
+        switch (hrException)
+        {
+        case HRESULT_FROM_WIN32(ERROR_TIMEOUT):
+        case WnpErrors::WNP_E_NOT_CONNECTED:
+        case WPN_E_OUTSTANDING_CHANNEL_REQUEST:
```

Comment:
Is WPN_E_OUTSTANDING_CHANNEL_REQUEST publicly available? Shouldn't it be part of WnpErrors?

Reply:
Never mind. Looks like it is:
https://docs.microsoft.com/zh-cn/windows/win32/com/com-error-codes-9

In the long term, it is possibly for the best that we also include some of these retryable errors in winerrors.h

Reply:
yeah it is available in winerrors.h